### DISCUSSION TITLE
Integration of video within OpenMCT ?

### DISCUSSION THREAD

**Discussion by linuxguy123** [nasa/openmct#3873] [Community Q&A]
Hi. 

We are looking at using OpenMCT for a project.  

We would really like to integrate video captured during a mission into MCT.    After some brainstorming we arrived at the following list of progressively harder "wants":

1) view live video in an MCT object  
2) view live video in an MCT object with object or texts overlays that display data in the video object
3) store video of a mission with the other data from a mission
4) be able to replay a mission session and have the video replay as well

Does anyone else have a similar want/need to use video within MCT ?   How are others integrating video within MCT ?

Has anyone started implementing video functionality in MCT ?

Thanks to the OpenMCT developers.  It looks fantastic.  We can't wait to start using it.

**Comment by linuxguy123** [comment]
I'm thinking of writing a video object for openMCT that uses video.js and videojs-record.   '

Thoughts ?

**Comment by jvigliotta** [comment]
Hi @linuxguy123, we don't have any specifically video based plugins for Open MCT, but we've had people utilize our [Web Page Plugin](https://github.com/nasa/openmct/tree/master/src/plugins/webPage) before to display videos. It'll take a url (could be where the video is hosted) and will plug it into an iframe. We've had some creative users actually use our [Imagery Plugin](https://github.com/nasa/openmct/tree/master/src/plugins/imagery) to feed in frames (around 5FPS) as incoming telemetry which was a pretty cool idea. I think your best bet is to utilize the Webpage Plugin or to create a plugin integrating videojs.

**Comment by linuxguy123** [comment]
As my description above states, the problem with displaying video (30 or 60Hz, maybe faster) in the web page plugin is that it doesn't store with the rest of the data and it isn't time stamped to coincide with data during playback.  Nor does it play back with the rest of the data. 

The project I'm working on needs video stored as well as data.    Any hints on how to implement this ?
